transaction-view/src/address_table_lookup_meta.rs

```diff
@@ -0,0 +1,167 @@
+use {
+    crate::{
+        bytes::{
+            advance_offset_for_array, advance_offset_for_type, check_remaining,
+            optimized_read_compressed_u16, read_byte,
+        },
+        result::Result,
+    },
+    solana_sdk::{packet::PACKET_DATA_SIZE, pubkey::Pubkey},
+};
+
+/// Contains meta-data about the address table lookups in a transaction packet.
+pub struct AddressTableLookupMeta {
+    /// The number of address table lookups in the transaction.
+    pub(crate) num_address_table_lookup: u16,
+    /// The offset to the first address table lookup in the transaction.
+    pub(crate) offset: u16,
+}
+
+impl AddressTableLookupMeta {
+    /// Get the number of address table lookups (ATL) and offset to the first.
+    /// The offset will be updated to point to the first byte after the last
+    /// ATL.
+    /// This function will parse each ATL to ensure the data is well-formed,
+    /// but will not cache data related to these ATLs.
+    pub fn try_new(bytes: &[u8], offset: &mut usize) -> Result<Self> {
+        // Read the number of ATLs at the current offset.
+        // Each ATL needs at least 34 bytes, so do a sanity check here to
+        // ensure we have enough bytes to read the number of ATLs.
+        const MIN_SIZED_ATL: usize = core::mem::size_of::<Pubkey>() + 1 + 1;
+        const MAX_ATLS_PER_PACKET: usize = PACKET_DATA_SIZE / MIN_SIZED_ATL;
+        // Maximum number of ATLs should be represented by a single byte,
+        // thus the MSB should not be set.
+        const _: () = assert!(MAX_ATLS_PER_PACKET & 0b1000_0000 == 0);
```

Comment:
I forgot about this for your other PRs, but Rust has `const_assert_eq` and `const_assert` (already available in our monorepo) that checks constraints during compile time. Perhaps you can use them here (and in all the other cases - feel free to do that in another PR if you like my idea).

Reply:
Haven't looked at the crate that provides those, but have been trying to keep dependencies to minimum.

Doesn't seem necessary except for slightly nicer syntax, but I don't feel the syntax here is that bad honestly. Maybe you/others feel differently?

Reply:
The nicer syntax is my preference, but this is also a nitpick. I'll leave it up to you. Perhaps, just ask a second opinion from the other reviewers.